### DISCUSSION TITLE
DateTimePicker date and time format

### DISCUSSION THREAD

**Discussion by VPKSoft** [picoe/Eto#2149] [Q&A] [answered]
Hi,
How can the date and/or time format be set to a DateTimePicker control? The unspecified format is something I have never seen (at least not being used with a Finnish region settings) 🙂
![image](https://user-images.githubusercontent.com/40712699/155588219-9cc3f42b-7385-49ee-b274-e24ca9777c03.png)
The settings:
![image](https://user-images.githubusercontent.com/40712699/155588615-801ac77b-8eb5-4f59-bef0-f0d58df227d8.png)
This is how the OS (W11) displays the date and time, so it is a 24-hour clock:
![image](https://user-images.githubusercontent.com/40712699/155588853-51370174-bb95-43ff-a2e6-6b4622679c64.png)

The DateTimePicker is specified as follows:
```cs
var dateTimePickerJump = new DateTimePicker { Mode = DateTimePickerMode.DateTime, Value = DateTime.Now };
```

Eto.Forms version 2.6.1, WPF-package in Windows.

Ps.
I'm a newbie to Eto.Forms, drafting a [2D sky chart](https://github.com/VPKSoft/StarMap2D)

**Comment by allsorts46** [comment]
If you're using the WinForms implementation, which from your screenshot it looks like you might be, the Eto.WinForms implementation of DateTimePicker tries to handle the formatting itself and is kind of broken. It doesn't support the full range of date/time format characters that the system one does, which might explain the weird display you have. It also has strange keyboard behaviour and clipping issues when the time string is too long.

Luckily it's easy to disable this custom behaviour, as there's a property on the class that implements the handler. Add this line somewhere in your app startup after you initialise Eto:

`Eto.Style.Add<Eto.WinForms.Forms.Controls.DateTimePickerHandler>(null, h => h.Control.ExtendedMode = false);`

**Reply by VPKSoft** [reply]
Thanks @allsorts46, this is WPF actually (I forgot to mention that - sorry):
![image](https://user-images.githubusercontent.com/40712699/155656179-01d7e893-1422-44c5-93d7-3229612db400.png)

An the version is 2.6.1 and the project file is just SDK, no explicit NuGet references:
```xml
<Project Sdk="Eto.Forms/2.6.1">
```

In a Linux box (Mint) the date seems to be in US-format even though it is set system-wide to be in Finnish, but that might be a .NET issue as other programs interpret the time format wrong as well. Running in Linux:
![image](https://user-images.githubusercontent.com/40712699/155659432-d81a2bba-9a3e-4f29-b399-ee7c969c3acc.png)

The length of the DateTimePicker doesn't seem to be the case as the same format is shown if I specify the width of the picker longer, it doesn't change. The format seems to be in Windows as localized AM/PM designator.

**Comment by VPKSoft** [comment, marked as answer]
Hi,
Setting the current thread UI locale, as defined below solved the formatting issue:
```cs
System.Threading.Thread.CurrentThread.CurrentUICulture = new CultureInfo("fi-FI");
```

So I'll just make it a setting or clone the value from:
```cs
System.Threading.Thread.CurrentThread.CurrentUICulture = System.Threading.Thread.CurrentThread.CurrentCulture;
```

Probably the latter is a better option.
